packager/mpd/base/representation.cc

```diff
@@ -208,6 +208,13 @@ void Representation::SetSampleDuration(uint32_t frame_duration) {
   }
 }
 
+void Representation::SetSegmentDuration() {
+  int64_t sd = mpd_options_.mpd_params.target_segment_duration * media_info_.reference_time_scale();
+  if (sd <= 0)
+    return;
```

Comment:
Do you need to log an error here? Or this error case can be expected?

Reply:
Good call. This case would only be hit if the segment duration is negative or 0. It seems that Packager does not check for negative segment durations anywhere else, which is an issue. I'll add an error to address this.